src/pybind/mgr/volumes/module.py

```diff
@@ -622,6 +640,22 @@ def _cmd_fs_subvolume_create(self, inbuf, cmd):
                                         mode=cmd.get('mode', '755'),
                                         namespace_isolated=cmd.get('namespace_isolated', False))
 
+    @mgr_cmd_wrap
```

Comment:
Instead of providing a way to create a subvolume with a particular version, the plugin could just disallow auto-upgrade to the most recent version. Wouldn't that suffice? As far as writing tests is concerned, the test case could hand-craft a particular subvolume (version).

Reply:
By default, the code always creates a subvolume of version `v2`. So disallowing an upgrade of a v2 subvolume is pointless.

I'd like to understand the aversion to the `fs subvolume create_version` CLI interface.
BTW, the `create_version` CLI does indeed get listed on the command-line help just as other commands are listed.

Reply:
> By default, the code always creates a subvolume of version `v2`. So disallowing an upgrade of a v2 subvolume is pointless.

Agreed.

> 
> I'd like to understand the aversion to the `fs subvolume create_version` CLI interface. BTW, the `create_version` CLI does indeed get listed on the command-line help just as other commands are listed.

That's the very reason I have a bit of an iffy against it.

Reply:
The hidden CLI commands we spoke about is a broader effort which should be proposed in a separate change. I propose for disabling auto-upgrade in this change for the sake of writing tests.

BTW, v1 subvolumes with snapshots should not get auto-upgraded. Are you not seeing this behaviour?

Reply:
> The hidden CLI commands we spoke about is a broader effort which should be proposed in a separate change. I propose for disabling auto-upgrade in this change for the sake of writing tests.

Okay, I'll see how I can hand-craft a v1 .meta file and run the tests.
I'll work with the non-persistent `allow_subvolume_upgrade` flag.

> 
> BTW, v1 subvolumes with snapshots should not get auto-upgraded. Are you not seeing this behaviour?

ack - v1 subvolumes with snapshots does not get auto-upgraded
I've confirmed this.